kite-data/kite-data-oozie/src/test/java/org/kitesdk/data/oozie/TestKiteURIHandler.java

```diff
@@ -215,6 +238,90 @@ public void existsForNonReadiableView() throws URIHandlerException, URISyntaxExc
+    uriHandler.exists(uri, null);
+
+    defaultConf = DefaultConfiguration.get();
+    Assert.assertEquals("test.value", defaultConf.get("test.property"));
+    Assert.assertEquals("something.else", Services.get().get(HCatAccessorService.class).getHCatConf().get("test.value"));
+  }
+
+  @Test
+  public void noConfigLoadedWhenNoServices() throws URIHandlerException, URISyntaxException {
+    URI uri = new URI("view:file:target/data/data/nomailbox?message=hello");
+    uriHandler.exists(uri, null);
+    Assert.assertNull(Services.get());
+  }
+
+  @Test
+  public void noConfigLoadedWhenNoHCatAccessor() throws URIHandlerException, URISyntaxException, ServiceException, FileNotFoundException, IOException {
+    setupHCatService(false, false);
+
+    URI uri = new URI("view:file:target/data/data/nomailbox?message=hello");
+    uriHandler.exists(uri, null);
+    Assert.assertNotNull(Services.get());
+    Assert.assertNull(Services.get().get(HCatAccessorService.class));
+  }
+
+  @Test
+  public void noConfigLoadedIfNoHCatAccessorConfig() throws URIHandlerException, URISyntaxException, ServiceException, FileNotFoundException, IOException {
+    setupHCatService(true, false);
+    URI uri = new URI("view:file:target/data/data/nomailbox?message=hello");
+    uriHandler.exists(uri, null);
+    Assert.assertNotNull(Services.get());
+    Assert.assertNotNull(Services.get().get(HCatAccessorService.class));
+    Assert.assertNull(Services.get().get(HCatAccessorService.class).getHCatConf());
+  }
+
+  private void setupHCatService(boolean loadHCat, boolean loadHCatConfig) throws ServiceException, FileNotFoundException, IOException {
+    serviceTempDir = Files.createTempDir();
+    File confDir = new File(serviceTempDir, "conf");
+    File hadoopConfDir = new File(confDir, "hadoop-conf");
+    File hadoopConfTarget = new File(hadoopConfDir, "hadoop-site.xml");
+    File actionConfDir = new File(confDir, "action-conf");
+    File hiveConfDir = new File(confDir, "hive-conf");
+    File hiveConfTarget = new File(hiveConfDir, "hive-site.xml");
+    File oozieSiteTarget = new File(confDir, "oozie-site.xml");
+    confDir.mkdir();
+    hadoopConfDir.mkdir();
+    actionConfDir.mkdir();
+    hiveConfDir.mkdir();
+
+    Configuration oozieSiteConf = new Configuration(false);
+
+    if(loadHCatConfig) {
+      oozieSiteConf.set("oozie.service.HCatAccessorService.hcat.configuration", hiveConfTarget.getAbsolutePath());
+    }
+
+    oozieSiteConf.set("oozie.services", "org.apache.oozie.service.HadoopAccessorService");
+    oozieSiteConf.writeXml(new FileOutputStream(oozieSiteTarget));
+    conf.writeXml(new FileOutputStream(hadoopConfTarget));
+
+    Configuration hiveConf = new Configuration(false);
+    hiveConf.set("test.property", "test.value");
+    hiveConf.writeXml(new FileOutputStream(hiveConfTarget));
+
+    // set to the temp directory
+    System.setProperty("oozie.home.dir", serviceTempDir.getAbsolutePath());
+
+    Services services = new Services();
```

Comment:
I took the route up on [line 113](https://github.com/kite-sdk/kite/pull/399/files#diff-bc98164dc4c52f3fd75c88e983b578eaR113) of calling the static "get" on the Services object to see if it had been initialized. On init it looks like the Services object just sets itself for singleton access. I then call destroy on the service object.

This is probably a bit harder to trace, it requires that you're familiar enough with the services object to catch that it's destroying the same object. It might be better for that reason just to keep a reference.